### DISCUSSION TITLE
Backing up libki (docker install)

### DISCUSSION THREAD

**Discussion by alanzaf** [Libki/libki-server#338] [Q&A]
Hi there,
I have installed libki as a docker container. I'd like to take a backup of the system (db included) and i've found a script for doing it under /app/scripts/utilities. When i try to run the script i get this:

root@9b6d2ca0117c:/app/script/utilities# bash backup.sh 

libki: unrecognized service
backup.sh: line 41: ps: command not found
mysqldump: Got error: 2002: "Can't connect to local MySQL server through socket '/run/mysqld/mysqld.sock' (2)" when trying to connect
cp: cannot stat '/etc/init.d/libki': No such file or directory
tar: libki-server: Cannot stat: No such file or directory
tar: Exiting with failure status due to previous errors
tar: init: Cannot stat: No such file or directory
tar: Exiting with failure status due to previous errors
libki: unrecognized service
backup.sh: line 81: ps: command not found

Can anybody give some help, please?

Thanks in advance.

**Comment by kylemhall** [comment]
That backup script was written with the assumption that mysql is installed on the same system as the Libki server. It looks like you are using docker containers so this script isn't going to work for you.

I would recommend using https://github.com/sixhop/AutoMySQLBackup which you can even run in a container ( https://github.com/selim13/docker-automysqlbackup ).

There's no need to backup your image as it's static. If you are mounting a volume with a config file you might want to back that up, but I recommend using environment variables instead!
